### PR TITLE
Fix UnresolvedAddressException connecting to job master

### PR DESCRIPTION
**Pull request by bf8086**
Sometimes the `InetSocketAddress` given to the client is not resolved, causing gRPC to panic when connecting to the server. This change adds a retry mechanism and force a resolution of the hostname before performing connect.

### CODE REVIEW

**Comment by bf8086** [comment]
@ggezer @calvinjia PTAL.

File: core/common/src/main/java/alluxio/AbstractClient.java [hunk 1]
```diff
@@ -212,6 +213,16 @@ public synchronized void connect() throws AlluxioStatusException {
...
         continue;
       }
+      if (mAddress.isUnresolved()) {
+        // Sometimes the acquired addressed wasn't resolved, retry resolving before
+        // using it to connect.
+        LOG.info("Retry resolving address {}", mAddress);
+        // Creates a new InetSocketAddress to force resolving the hostname again.
+        mAddress = new InetSocketAddress(mAddress.getHostName(), mAddress.getPort());
+        if (mAddress.isUnresolved()) {
+          LOG.warn("Failed to resolve address on retry {}", mAddress);
```
Comment (ggezer) [inline core/common/src/main/java/alluxio/AbstractClient.java]: Maybe a little sleep for giving network stack a chance?

Comment (bf8086) [reply core/common/src/main/java/alluxio/AbstractClient.java]: The retry policy will make sure the client sleeps before trying another resolution.

**Comment by calvinjia** [comment]
alluxio-bot, merge this please.

**Pull request merged**